doc/src/learn/tokenomics/gas-in-sui.md

```diff
@@ -57,21 +57,23 @@ All transactions need to be submitted together with a Gas Budget. This provides
 
 A transaction’s Gas Budget is defined in SUI units and transactions will be successfully executed if:
 
-`gas_budget >= total_gas_fees`
+`gas_budget >= max{computation_fees,total_gas_fees}`
 
-If the Gas Budget does not fulfill this condition — and thus is insufficient to cover a transaction’s gas fees — then the transaction will fail and the entire Gas Budget will be charged. Importantly, the minimum Gas Budget is 2000 MIST. This ensures validators can be compensated with at least 2000 MIST even if the Gas Budget is incorrectly specified and the transaction aborts. Additionally, this protects the Sui Network from being spammed with a large number of transactions with minimal gas budgets. The maximum Gas Budget is 50 billion MIST or 50 SUI. This protects the network against overflow of internal multiplications and gas limits for denial of service attack.
+If the Gas Budget does not fulfill this condition then the transaction will fail and the entire Gas Budget will be charged. Ultimately, a successful transaction will require the end user to pay the transaction's `total_gas_fees`. However, since it is challenging to perfectly forecast computation time before the transaction is processed, the `gas_budget` condition also requires the `gas_budget` to be at least as large as the transaction's `computation_fees` in case the transaction aborts. In some cases -- especially in the presence of high storage rebates, and, thus negative net storage fees -- the Gas Budget may be higher than the total gas fees paid by the end user.
```

Comment:
> If the Gas Budget does not fulfill this condition then the transaction will fail and the entire Gas Budget will be charged.

This is slightly inaccurate. If we fail while charging for non-input storage mutation, we won't charge the entire gas budget - only the gas actually used (computation cost + input object mutation due to version # change and gas object mutation) will be charged.

Reply:
thanks for flagging! i've update the text can you check?